### PR TITLE
[importer] auto formatting table and header name

### PR DESCRIPTION
**Pull request by agl29**
## What changes were proposed in this pull request?

- Changing all char to '_' except alphanumeric and '-'
- Hive and Impala don't support any other char except alphanumeric and '-'

## How was this patch tested?

- Unit test
- Manual

### CODE REVIEW

**Comment by Harshg999** [comment]
Most probably need to do the Py2 latest branch for new values change again? @agl29

File: desktop/libs/indexer/src/indexer/api3.py [hunk 1]
```diff
@@ -239,7 +240,7 @@ def guess_field_types(request):
 
       if file_format['format']['hasHeader']:
         sample = csv_data[1:5]
-        column_row = csv_data[0]
+        column_row = [re.sub('[^0-9a-zA-Z]+', '_', col) for col in csv_data[0]]
```
Comment (Harshg999) [inline desktop/libs/indexer/src/indexer/api3.py:243]: Maybe declare this regex above and use the variable in diff places? Easier to modify it at one place if needed

File: desktop/libs/indexer/src/indexer/api3_tests.py [hunk 2]
```diff
@@ -69,7 +69,7 @@ def test_xlsx_local_file_upload():
 ,scattered,,
 '''
   with open(BASE_DIR + '/apps/beeswax/data/tables/book.xlsx', 'rb') as file:
-    uploaded_file = InMemoryUploadedFile(file=file, field_name='test', name='book.xlsx', content_type='xlsx', size=100, charset='utf-8')
+    uploaded_file = InMemoryUploadedFile(file=file, field_name='test', name='book(1).xlsx', content_type='xlsx', size=100, charset='utf-8')
```
[the remark this answers is not in the document]
Comment (agl29) [reply desktop/libs/indexer/src/indexer/api3_tests.py:72]: as commit says we are formatting the table name automatically so I am giving a faulty name then checking here `assert_true("book_1_xlsx" in path)`

[the remark this answers is not in the document]
Comment (Harshg999) [reply desktop/libs/indexer/src/indexer/api3_tests.py:72]: So we are opening and uploading `book.xlsx` but we are renaming it as `books(1).xlsx` which then turns to `book_1_xlsx`?

[the remark this answers is not in the document]
Comment (agl29) [reply desktop/libs/indexer/src/indexer/api3_tests.py:72]: yes, I don't want to change the name in the directory that's why I am using it this way.

File: desktop/libs/indexer/src/indexer/api3.py (hunk 1 above)
Comment (agl29) [reply desktop/libs/indexer/src/indexer/api3.py:243]: Actually, external tables (we are using in remote file option) support some more charsets so I am not using this for the remote_file option as there are more things to understand is this needed or not in other options. In the future, if needed we can modify it accordingly?

**Review by Harshg999 — APPROVED** [review]

**Pull request merged**